core/src/main/java/org/testcontainers/containers/GenericContainer.java

```
@@ -473,6 +476,7 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
         Map<String, String> labels = createCommand.getLabels();
         labels = new HashMap<>(labels != null ? labels : Collections.emptyMap());
         labels.putAll(DockerClientFactory.DEFAULT_LABELS);
+        labels.putAll(this.labels);
```

I would do it **before** `DockerClientFactory.DEFAULT_LABELS` (they must be set to the values we provide), and also before `createContainerCmdModifiers`

---

I a developer really wants to overwrite the default labels he or she should be able to do so.
I tried overwriting the `org.testcontainers` label with false and left the `session_id` untouched - the containers still were removed.

---

sorry, IMO this is too dangerous. 
Also, why would anyone override internal labels of Testcontainers?

---

Ok - should we silently ignore `org.testcontainers.*` labels or throw an exception?

---

an exception, plus an order change ( `createContainerCmdModifiers` are supposed to be low level and should be able to override any pre-set values)

---

As the university is closing down - I'll change this tomorrow.

---

Sure! Thanks for your contribution :)

---

@bsideup done :) Please review all the changes, I will squash the commits when you give your okay ;)

---

Thanks! No need to squash, GitHub allows us to do it automatically 🎉